pinot-segment-local/src/main/java/org/apache/pinot/segment/local/startree/v2/builder/StarTreeIndexSeparator.java

```diff
@@ -57,10 +61,15 @@ public StarTreeIndexSeparator(File indexMapFile, File indexFile, List<StarTreeV2
     int numStarTrees = starTreeMetadataList.size();
     _builderConfigList = new ArrayList<>(numStarTrees);
     _numDocsList = new ArrayList<>(numStarTrees);
+    StringBuilder logExistingStarTrees = new StringBuilder();
+    logExistingStarTrees.append("Existing star-tree configs :");
     for (StarTreeV2Metadata starTreeMetadata : starTreeMetadataList) {
-      _builderConfigList.add(StarTreeV2BuilderConfig.fromMetadata(starTreeMetadata));
+      StarTreeV2BuilderConfig config = StarTreeV2BuilderConfig.fromMetadata(starTreeMetadata);
+      _builderConfigList.add(config);
+      logExistingStarTrees.append("\n").append(config);
       _numDocsList.add(starTreeMetadata.getNumDocs());
     }
+    LOGGER.debug(logExistingStarTrees.toString());
```

Comment:
Same here. Check whether debug log is enabled, and construct the log from `_builderConfigList` only when it is enabled